### DISCUSSION TITLE
Various Client Issues

### DISCUSSION THREAD

**Discussion by huckabarry** [akirk/enable-mastodon-apps#19] [General]
Apologies if I'm not understanding fully, hoping someone can help explain. Below are my issues/questions. Maybe this would've been better as a discussion item but I didn't see a discussion area to post this. 

I was able and to login to my Wordpress site with Ivory, Metatext, Elk (originally - more below), Mona (Beta), and Ice Cubes but not Mammoth or the official Mastodon app. There are issues for sure, and some are the same for all of them while some have unique issues. 

1. Unlike looking at my WP profile from another Mastodon account, I very often can't see my own bio or my own friends lists that I follow or who follow me from any of the apps. I've tried this from several outside accounts and although any bio formatting or updates can be delayed in presenting the updated info, I can usually see what is there from an outside Mastodon account but never when logged in from my WP account into any of the apps. 
2. I cannot see anyone else's posts from all but 2 (Mona/Ivory) in the feed, and those were are very delayed. I only see my own posts and never anyone I follow through the Friends plugin from most of the apps. I am the only person who shows up within my feed generally. 
3. There is no way to edit your own bio, header image, or avatar. This is less critical, but throws errors. 
4. You cannot click on other people's profiles, can't star or boost anyone's posts but your own, cannot follow anyone. Even in those apps that allow you to see the feeds of other instances, the apps do not allow you to interact with posts from other accounts in any way whatsoever.
5. Ivory logs in and I am able to see momentarily that people I'm subscribed to show up in my feed but Ivory almost immediately crashes now every time I open my WP account. 
6. Ice Cubes shows no bio, shows 0 followers and 0 following, shows zero posts of my own, on my profile page. I can see some of my posts from yesterday in the timeline. When I'm logged into a real Mastodon account and go over to my WP profile, I can see my bio, my largest posts, and the number of people who are following me and who I'm following. 
7. Elk has started throwing and error when logging in:
"{"error":"invalid_client","error_description":"The client id supplied is invalid"}"
8. Mona Beta logs in, shows followers in the feed, but does not show the appropriate info on my own profile despite being shown when logged into a real Mastodon account and looking at it. It also doesn't show my latest posts. This app has very similar behavior to Ivory. 
9. Metatext logs in but only shows I have 3 followers and nothing else. The feed shows updates from a day ago but not my latest. 
10. Mammoth and the official Mastodon apps will not login at all. They allow you to go to the WP login, allow you to click the "authorize" button, leave their breadcrumbs in the Mastodon Apps plug-ins authorizations list, but do not complete the process of added the account. You can do the authorization a hundred times and it will add 100 authorizations and tokens to
The lists, not won't add the account.

So my question is, what is the purpose of this app? Is it just to post status updates from a native Mastodon app and that is all, because that is essentially the only function that's working for me from any of the various apps. 

If that is the case, it does not offer a better experience than updating statuses from Wordpress. 

Hoping these are just bugs because it's so new because I had super high interest in this project and in fact was one of the only reasons I began migrating from Micro.blog to Wordpress.

**Comment by akirk** [comment]
Hi, I've indeed moved this to a discussion. Let me preface this that there is a reason of this plugin being in version zero point something. This is not yet feature complete or assumed to be mostly bug free.

Implementing the Mastodon API in WordPress has proven to be tricky, many apps have their unique quirks or insist on some specific JSON return values.

That said, the main goal is to be able to use Mastodon apps with your own WordPress, either in combination with the Friends+ActivityPub but also standalone which works well when you want to use Mastodon apps to post status like messages to a blog (and seeing everyone's posts on the same blog can also be useful when multiple people log things together).

To address your points:

> 1. [...] I very often can't see my own bio or my own friends lists that I follow or who follow me from any of the apps. 

Could you please be a bit more specific? When I view other people from inside a Mastodon app, I do see their profile, avatar and picture. The posts, followers and following numbers are often 0. This still needs a more stringent implementation.

> 2. [...]  I only see my own posts and never anyone I follow through the Friends plugin from most of the apps. I am the only person who shows up within my feed generally.

I cannot confirm this, it works well for me. Did you check your Friends page on your WordPress blog? Are the posts equally delayed there?

> 3. There is no way to edit your own bio, header image, or avatar.

Yes, it's not implemented (yet). See the project's README.md.

> 4. You cannot click on other people's profiles, can't star or boost anyone's posts but your own, cannot follow anyone. Even in those apps that allow you to see the feeds of other instances, the apps do not allow you to interact with posts from other accounts in any way whatsoever.

I cannot confirm this either. I can do all of this. It sounds like something is not quite right with your install. Could you please check your error log to see if you find anything that seems related?

> 5. Ivory logs in and I am able to see momentarily that people I'm subscribed to show up in my feed but Ivory almost immediately crashes now every time I open my WP account.

I heard about this and I am trying to collaborate with others in #11 to fix the crash.

> 6. Ice Cubes shows no bio, shows 0 followers and 0 following, shows zero posts of my own, on my profile page.

Hm, I can view my own bio fine. Maybe some of the fixes in 0.3.4 helped?

> 7. Elk has started throwing and error when logging in

Maybe you deleted that app entry in your backend? That might explain it if it worked before.

> 8. Mona Beta logs in, shows followers in the feed, but does not show the appropriate info on my own profile despite being shown when logged into a real Mastodon account and looking at it.

Mona works pretty fine for me, except not showing the right number of "Following". I'll take a look at that but otherwise it works sufficiently well for me.

> 9. Metatext logs in but only shows I have 3 followers and nothing else. The feed shows updates from a day ago but not my latest

I just tested Metatext and it works quite fine, "just" the list of toots for myself or a person, logged it in #20.

> Mammoth and the official Mastodon apps will not login at all.

I think this should have been fixed in 0.3.3. The official Mastodon Android app worked before.

Please let me know if the new version improves things for you and please help us find and fix the bugs in the appropriate issues. Thanks!

**Reply by Happyfeet01** [reply]
Hello,

Which App is currently recommend to interact with followers?
